src/pull_request.py

```diff
@@ -0,0 +1,237 @@
+        """
+        logging.info("create new branch %s", branch_name)
+        try:
+            self._git_repo.git.checkout("-b", branch_name)
+            self._git_repo.git.add(".")
+            self._git_repo.git.commit("-m", f"'{commit_msg}'")
+            self._git_repo.git.push("-u", "origin", branch_name)
+        except GitCommandError as exc:
+            raise RepositoryClientError(f"Unexpected error creating new branch. {exc=!r}") from exc
+
+    def create_github_pull_request(self, branch_name: str, base: str) -> str:
+        """Create a pull request from given branch to base.
+
+        Args:
+            branch_name: Branch name from which the pull request will be created.
+            base: Base branch to which the pull request will be created.
+
+        Raises:
+            RepositoryClientError: if unexpected error occurred during git operation.
+
+        Returns:
+            The web url to pull request page.
+        """
+        logging.info("create pull request %s", branch_name)
+        try:
+            pull_request = self._github_repo.create_pull(
+                title=ACTIONS_PULL_REQUEST_TITLE,
+                body=ACTIONS_PULL_REQUEST_BODY,
+                base=base,
+                head=branch_name,
+            )
+        except GithubException as exc:
+            raise RepositoryClientError(
+                f"Unexpected error creating pull request. {exc=!r}"
+            ) from exc
+
+        return pull_request.html_url
+
+    def is_dirty(self) -> bool:
+        """Check if repository path has any changes including new files.
+
+        Returns:
+            True if any changes have occurred.
+        """
+        return self._git_repo.is_dirty(untracked_files=True)
+
+    def get_active_branch(self) -> str:
+        """Get name of currently active branch on local git repository.
+
+        Returns:
+            Name of currently active branch.
+        """
+        return self._git_repo.active_branch.name
+
+    def set_active_branch(self, branch_name: str) -> None:
+        """Set current active branch to an given branch that already exists.
+
+        Args:
+            branch_name: target branch that already exists in git.
+        """
+        self._git_repo.git.checkout(branch_name)
+
+
+def create_pull_request(repository: RepositoryClient) -> str:
+    """Create pull request for changes in given repository path.
+
+    Args:
+        repository: A git client to interact with local and remote git repository.
+
+    Raises:
+        InputError: if pull request branch name is invalid or the a branch
+        with same name already exists.
+
+    Returns:
+        Pull request URL string. None if no pull request was created/modified.
+    """
+    base = repository.get_active_branch()
+    if base == DEFAULT_BRANCH_NAME:
+        raise InputError(
+            f"Pull request branch cannot be named {DEFAULT_BRANCH_NAME}."
```

Comment:
Should the reason it can't be called this be explained here? E.g., because that is the branch that will be created for the migration?

Reply:
Added an explanation about reserved branch name:
```
raise InputError(
            f"Pull request branch cannot be named {DEFAULT_BRANCH_NAME}."
            f"Branch name {DEFAULT_BRANCH_NAME} is reserved for creating a migration branch."
            "Please try again after changing the branch name."
        )
```

Thank you for the comment!